Firestore/src/support/secure_random_bsd.cc

```diff
@@ -0,0 +1,31 @@
+/*
```

Comment:
There will be at least `secure_random_windows.cc` which does a whole dance to get to `RtlGenRandom` and `secure_random_ssl.cc` which delegates to boringssl.

These are each very different implementations and while you can put these in a single file, it gets gross.

My expectation is that we constrain this kind of platform switching to files in `src/support`. Throughout `src/core` we'll end up with matching `.h` and `.cc` files.

If it would make you feel better we could make this `secure_random_arc4random.cc`.

Reply:
_arc4random feels more natural to me (vs _bsd) since this should apply to any system that supports arc4random. But I don't feel strongly about it.

Reply:
So for my own understanding, are we going to have some secure_random_xyz.cc implementation that works on Android too?  I know we're not planning to replace our "mainstream" Android client with C++, but I'm fuzzy on what platforms we're targeting in general.  iOS, Windows, ...?

Reply:
Yes. I've already written it too! :-)

It's `secure_random_boringssl.cc`. Forthcoming, but not yet written is `secure_random_windows.cc`.

Note that we need a separate windows one because boringssl doesn't support building with anything older than VS 2015. I don't want to just use openssl for this though: openssl defines `RAND_bytes` in a way that can fail if there's insufficient entropy. Boringssl won't fail in this way.

Reply:
Nice :-)